server/controllers/availability.js

```diff
@@ -0,0 +1,104 @@
+const Availability = require("../models/availability");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+// @route POST /schedule/
+// @desc Create a schedule
+// @access Public
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const profile = await User.findById(req.user.id, "profile");
+
+  if (!profile) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+  const profileId = await Profile.findOne({ userId: req.user.id });
+  const id = profileId._id;
```

Comment:
```suggestion
  const profileId = profile._id;
```
You can change this throughout

Reply:
changed from id to ProfileId and profileId to profile throughout the controller